libmamba/src/core/package_fetcher.cpp

```diff
@@ -436,6 +436,20 @@ namespace mamba
 
         nlohmann::json repodata_record = m_package_info;
 
+        // For explicit spec files (URLs), m_package_info has empty depends/constrains arrays
+        // that would overwrite the correct values from index.json. Remove these empty fields.
+        auto depends_it = repodata_record.find("depends");
+        auto constrains_it = repodata_record.find("constrains");
+
+        if (depends_it != repodata_record.end() && depends_it->empty())
```

Comment:
Since `depends_it`  is not used elsewhere, you can initialize it in the if statement to limit its scope.